### DISCUSSION TITLE
Allow `require_auth` from non-existent accounts

### DISCUSSION THREAD

**Discussion by kalepail** [stellar/stellar-protocol#1464] [Core Advancement Proposals]
It's unfortunate that the `require_auth` and `require_auth_for_args` methods can only be called on Stellar account's that have been created on the network. It would be nice to be able to utilize existing wallet infrastructure for signing auth entries with account's that haven't been created on the network.

It definitely took me by surprise that this wasn't possible. I would love to see general auth entry signing be possible with accounts that haven't been created on the network.

**Comment by dmkozh** [comment]
It is possible to make auth work for non-existent accounts. These would be verified just using the account id as a public ed25519 key.

Note, that due to the SAC semantics of using trustlines/account entries for the classic account balances, non-existent accounts will not be able to hold any SAC balances, including XLM balance. They will be compatible with any contracts that don't use SAC.

If there is more generic demand for supporting stateless accounts, there are ways to achieve that as well in a more generic fashion. However, that's much more involved, so it would make sense to proceed just with the classic account support.

**Comment by namankumar** [comment]
> It's unfortunate that the `require_auth` and `require_auth_for_args` methods can only be called on Stellar account's that have been created on the network. It would be nice to be able to utilize existing wallet infrastructure for signing auth entries with account's that haven't been created on the network.
> 
> It definitely took me by surprise that this wasn't possible. I would love to see general auth entry signing be possible with accounts that haven't been created on the network.

Help me see why this is important with an example?

**Reply by kalepail** [reply]
I don't want to pay to create an account just to be able to sign a Soroban auth entry. It's really just a cost thing. It makes key rotation free, barrier to entry free, signing account can now just be ephemeral keypairs vs created keypairs. All of this seems good. It's not a use case it's an ease of use, friction reduction. Also it just seems obvious. As it is it's surprising. I don't like that. It should behave more obviously imo.
